application/F3DStarter.cxx

```diff
@@ -700,6 +708,123 @@ void F3DStarter::Render()
   f3d::log::debug("Render done");
 }
 
+//----------------------------------------------------------------------------
+void F3DStarter::SaveScreenshot(const std::string& filenameTemplate)
+{
+
+  const auto getScreenshotDir = []()
+  {
+    for (const char* const& candidate : { "XDG_PICTURES_DIR", "HOME", "USERPROFILE" })
+    {
+      char* val = std::getenv(candidate);
+      if (val != nullptr)
+      {
+        std::filesystem::path path(val);
+        if (std::filesystem::is_directory(path))
+        {
+          return path;
+        }
+      }
+    }
+
+    return std::filesystem::current_path();
+  };
+
+  const auto applyFilenameTemplate = [&](const std::string& fn_template)
+  {
+    std::string filename = fn_template;
+
+    {
+      std::regex re("\\{app\\}");
+      filename = std::regex_replace(filename, re, F3D::AppName);
+    }
+    {
+      std::regex re("\\{version\\}");
+      filename = std::regex_replace(filename, re, F3D::AppVersion);
+    }
+    {
+      std::regex re("\\{model\\}");
+      if (std::regex_search(filename, re))
+      {
+        const auto file = this->Internals->FilesList[this->Internals->CurrentFileIndex];
+        filename = std::regex_replace(filename, re, file.filename().string());
+      }
+    }
+
+    return std::filesystem::path(filename);
+  };
+
+  const auto findAvailableFilename = [&](const std::string& idealFilename)
+  {
+    const auto appendFilenameNumber = [](const std::filesystem::path& path, size_t n)
+    {
+      std::stringstream ss;
+      const std::string fn = path.string();
+      const std::string::size_type i = fn.find_last_of('.');
+      if (i != std::string::npos)
+      {
+        ss << fn.substr(0, i) << "_" << n << fn.substr(i);
+      }
+      else
+      {
+        ss << fn << "_" << n;
+      }
+      return ss.str();
+    };
+
+    std::string fn = idealFilename;
+    for (size_t i = 2; std::filesystem::exists(fn) && i < 1e8; ++i)
```

Comment:
(in a short loop its fine)

Reply:
why do we start at 2 though ?

Reply:
I'm not adding a number if the filename is already available (which would be `1`)